### PR TITLE
WIP: specifying resources in the hostfile

### PR DESCRIPTION
**Pull request by jmontoyam**
Hi @mrocklin,
what do you think about the proposed solution?

### CODE REVIEW

File: distributed/deploy/ssh.py [hunk 1]
```diff
@@ -256,8 +264,9 @@ def __init__(self, scheduler_addr, scheduler_port, worker_addrs, nthreads = 0, n
 
         # Start worker nodes
         self.workers = []
-        for i, addr in enumerate(worker_addrs):
-            self.add_worker(addr)
+        for addr, nthreads, nprocs in zip(worker_addrs, self.threads,
+                                          self.nprocs):
+            self.add_worker(addr, nthreads, nprocs)
```
Comment (mrocklin) [inline distributed/deploy/ssh.py]: Ideally we would make this general to all keywords.  I expect users to become confused why only certain keywords are respected in this way.

Comment (jmontoyam) [inline distributed/deploy/ssh.py]: Thanks for the feedback @mrocklin, what you mean is that I should take into account that the user might want to use the hostfile to specify any feature of the worker?, right?

Comment (mrocklin) [inline distributed/deploy/ssh.py]: Yes, that it what I mean.  Or, if not all features, it should at least be pretty trivial to extend this to almost all features, ideally without a long chain of if-else statements.  I hesitate to add any user-facing feature where some things work and some very related things don't work.  This tends to fill my inbox with confusing user reports :)

**Comment by mrocklin** [comment]
Closing as stale.  Please reopen if work picks up again

**Pull request closed without merging**